sympy/polys/factortools.py

```diff
@@ -125,6 +126,9 @@ def dmp_trial_division(f, factors, u, K):
             else:
                 break
 
+        if k == 0:
+            raise RuntimeError("trial division failed")
+
```

Comment:
This kind of threw me. I guess when the docstring refers to "factors" that means "known factors," i.e. we know that each one really does divide `f` at least once.

Is it important to raise an error, instead of just returning `0` as a multiplicity?

If so, maybe the docstring could say "known factors," or, if that's a puzzling phrase, just add something like "(Each factor must actually divide f at least once.)" at the end.

Reply:
We could remove this check. In principle these functions could be used with non-factors just as generic routines for finding multiplicity with candidate factors. In practice in the sympy codebase these functions are only ever called with genuine factors (multiplicity >= 1) and this is a very cheap check for something going wrong somewhere in the more complicated algorithms that produce the irreducible factors.

I've added the check to `dup_trial_division` as well.

Reply:
Sounds good.